apiserver/facades/agent/secretsmanager/secrets.go

```diff
@@ -57,8 +57,47 @@ type SecretsManagerAPI struct {
 	crossModelState CrossModelState
 }
 
+// SecretsManagerAPIV1 the secrets manager facade v1.
+// TODO - drop when we no longer support juju 3.1.0
+type SecretsManagerAPIV1 struct {
+	*SecretsManagerAPI
+}
+
+// GetSecretStoreConfig is for 3.0.x agents.
+// TODO - drop when we no longer support juju 3.0.x
+func (s *SecretsManagerAPIV1) GetSecretStoreConfig() (params.SecretBackendConfig, error) {
+	cfgInfo, err := s.GetSecretBackendConfig()
+	if err != nil {
+		return params.SecretBackendConfig{}, errors.Trace(err)
+	}
+	return cfgInfo.Configs[cfgInfo.ActiveID], nil
+}
+
 // GetSecretBackendConfig gets the config needed to create a client to secret backends.
-func (s *SecretsManagerAPI) GetSecretBackendConfig(arg params.SecretBackendArgs) (params.SecretBackendConfigResults, error) {
+// TODO - drop when we no longer support juju 3.1.x
+func (s *SecretsManagerAPIV1) GetSecretBackendConfig() (params.SecretBackendConfigResultsV1, error) {
+	cfgInfo, err := s.backendConfigGetter(nil)
+	if err != nil {
+		return params.SecretBackendConfigResultsV1{}, errors.Trace(err)
+	}
+	result := params.SecretBackendConfigResultsV1{
+		ActiveID: cfgInfo.ActiveID,
+		Configs:  make(map[string]params.SecretBackendConfig),
+	}
+	for id, cfg := range cfgInfo.Configs {
+		result.ControllerUUID = cfg.ControllerUUID
+		result.ModelUUID = cfg.ModelUUID
+		result.ModelName = cfg.ModelName
+		result.Configs[id] = params.SecretBackendConfig{
+			BackendType: cfg.BackendType,
+			Params:      cfg.Config,
+		}
+	}
+	return result, nil
+}
+
```

Comment:
Do we need to mask on SecretsManagerAPIV1 to ensure that `GetSecretBackendConfigs` does not  exist on SecretsManagerAPIV1?

Reply:
just to be safe, sure